### PR TITLE
Das 160 load cif

### PR DESCRIPTION
**Pull request by rozyczko**

### CODE REVIEW

File: App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml [hunk 1]
```diff
@@ -60,12 +61,53 @@ ColumnLayout {
...
+                        Generic.Variables.homePageFinished = true
+                        Generic.Variables.dataPageFinished = true
+                        Generic.Variables.samplePageFinished = true
+                        Generic.Variables.analysisPageFinished = Generic.Variables.isDebug ? true : false
+                        Generic.Variables.summaryPageFinished = Generic.Variables.isDebug ? true : false
+
+                        // The remove button will have to be enabled once we start actually adding phases
+                        //removeButton.enabled = true
+                    } else {
+                        failOpenDialog.visible = true
```
Comment (wardsimon) [inline App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml]: failOpenDialog doesn't exist. The following is needed, or a generic fail dialog
```
            Dialog {
                id: failOpenDialog
                parent: Overlay.overlay
                anchors.centerIn: parent
                modal: true
                opacity: 0.9
                visible: false
                Label {
                    id: infoLabel
                    anchors.centerIn: parent
                    text: 'Warning: File was not a valid `cif` file.'
                    color: "black"
                    font.family: Generic.Style.introThinFontFamily
                    font.pointSize: Generic.Style.systemFontPointSize + 1
                }
            }
```

File: Tests/pytest/App/PyImports/Models/test_FileStructureModel.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    # assure _setModelFromProject got called
+    assert m._data_model.rowCount() == 2
+    assert m._data_model.columnCount() == 1
+
+    role_1 = Qt.UserRole + 1
+    role_2 = Qt.UserRole + 2
+
+    assert len(m._data_model.roleNames()) == 2
+    assert m._data_model.roleNames()[role_1] == b"phasesRole"
+    assert str(m._data_model.roleNames()[role_2]) == "b'experimentsRole'"
```
Comment (wardsimon) [inline Tests/pytest/App/PyImports/Models/test_FileStructureModel.py]: Personally I'd prefer named roles e.g. phaseRole and expRole for example. But, minor comment

File: App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml [hunk 3]
```diff
@@ -60,12 +61,53 @@ ColumnLayout {
...
+                    }
+                GenericAppContentAreaButtons.Export {
+                    enabled: false;
+                    text: "Export selected phase to CIF"
+                    }
+            }
+            // Open project dialog
+            Dialogs1.FileDialog{
+                id: fileDialogLoadPhase
+                nameFilters: [ "CIF files (*.cif)", "Project files (*.zip)"]
```
Comment (wardsimon) [inline App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml]: Change to CIF files only.

File: App/PyImports/Calculators/CryspyCalculator.py [hunk 4]
```diff
@@ -24,29 +27,94 @@ def __init__(self, main_rcif_path, parent=None):
...
+
+        # This will notify the GUI models changed
+        self.projectDictChanged.emit()
+
+    def replacePhaseInSegment(self, segment, new_phase_name):
+        """
+        Replaces original phase name with the given name in a segment
+        """
+        segment_content = self._parseSegment(segment)
+        old_phase_name = self._phase_name
```
Comment (wardsimon) [inline App/PyImports/Calculators/CryspyCalculator.py]: `self._phase_name` is not defined in init

**Comment by wardsimon** [comment]
I wasn't sure if to submit review now (I.e could you see my comments). Will do second review after the bugs have been fixed

File: App/PyImports/Calculators/CryspyCalculator.py (hunk 4 above)
Comment (rozyczko) [reply App/PyImports/Calculators/CryspyCalculator.py]: it is now

File: App/PyImports/ProjectSentinel.py [hunk 5]
```diff
@@ -20,6 +20,31 @@ def __init__(self, parent=None):
...
+        self.experiment_rcif_path = None
+
+    @Slot(str)
+    def loadPhases(self, phases_rcif_path):
+        """
+        Load a structure from a file.
+        :param structure_rcif_path: URI to structure (r)cif file
+        :return:
+        """
+        self.phases_rcif_path = self.generalizePath(phases_rcif_path)
```
[the remark this answers is not in the document]
Comment (rozyczko) [reply App/PyImports/ProjectSentinel.py]: it is using it...
line 89 in Controls.qml
`projectControl.loadPhases(fileUrl)`

File: App/QmlImports/easyAnalysis/App/ContentArea/MainArea/Pages/SampleModel/TextView.qml [hunk 6]
```diff
@@ -6,27 +6,25 @@ import easyDiffraction 1.0 as Specific
...
+    property bool showContent: false
 
-    ScrollView {
-        anchors.fill: parent
+    ListView {
+        width: parent.width
+        height: parent.height
+        boundsBehavior: Flickable.StopAtBounds
+        model: Specific.Variables.cif
+        ScrollBar.vertical: ScrollBar { policy: ScrollBar.AsNeeded }
```
[the remark this answers is not in the document]
Comment (rozyczko) [reply App/QmlImports/easyAnalysis/App/ContentArea/MainArea/Pages/SampleModel/TextView.qml]: No idea how to fix it :( I'll create a bug report in jira and will get some help from the pros.

File: Tests/pytest/App/PyImports/Models/test_FileStructureModel.py (hunk 2 above)
Comment (rozyczko) [reply Tests/pytest/App/PyImports/Models/test_FileStructureModel.py]: as in renaming temporary variables `role_1` to `phaseRole`?

Comment (rozyczko) [reply Tests/pytest/App/PyImports/Models/test_FileStructureModel.py]: done

File: App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml (hunk 3 above)
[the remark this answers is not in the document]
Comment (wardsimon) [reply App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml]: `nameFilters: [ "CIF files (*.cif)", "Project files (*.zip)"]` should be `nameFilters: [ "CIF files (*.cif)"]`

[the remark this answers is not in the document]
Comment (wardsimon) [reply App/QmlImports/easyAnalysis/App/ContentArea/Sidebar/Pages/SampleModel/StructureView/Controls.qml]: Hmm, on my version this is done. Sorry :-)

**Pull request merged**